core/jazz_es-kinesis-log-streamer/components/utils.js

```diff
@@ -136,93 +156,99 @@ function getApiLogsData(payload) {
   return bulkRequestBody;
 }
 
-function getLambdaLogsData(payload) {
-  let bulkRequestBody = '',
+function getLambdaLogsData(payload, callback) {
+  getLogsGroupsTags(payload.logGroup)
+  .then((tagsResult) => {
+    let bulkRequestBody = '',
     data = {}
-  data.asset_type = "lambda";
-  data.provider = "aws";
-  let domainAndservice, serviceInfo;
-  data.request_id = getInfo(payload.logEvents, config.PATTERNS.lambda_request_id);
-  if (data.request_id) {
-    data.asset_identifier = payload.logGroup.split(config.PATTERNS.asset_identifier_key)[1];
-    let environment = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment, 2);
-		if (environment === "dev") {
-			let dev_environment = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment_dev, 2);
-			serviceInfo = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment_dev, 1);
-			data.environment = dev_environment;
-		} else {
-			data.environment = environment;
-			serviceInfo = getSubInfo(payload.logGroup, config.PATTERNS.lambda_domain_service, 1);
-		}
-
-		domainAndservice = serviceInfo;
-		if(serviceInfo.indexOf(config.PATTERNS.sls_app_function) !== -1 ) { // if yes then sls-app function
-			let serviceInfoArr = serviceInfo.split(config.PATTERNS.sls_app_function);
-			domainAndservice = serviceInfoArr[0];
-		} 
-
-		logger.debug("domainAndservice: " + domainAndservice)
-    let domain = domainAndservice.substring(0, domainAndservice.indexOf("_"));
-    if (domain) {
-      data.domain = domain;
-      data.servicename = domainAndservice.substring(domain.length + 1);
-    } else {
-      data.domain = "";
-      data.servicename = domainAndservice;
-    }
-
-    if (data.servicename) {
-      payload.logEvents.forEach(function (logEvent) {
-
-        data.request_id = getSubInfo(logEvent.message, config.PATTERNS.guid_regex, 0);
-
-        data.platform_log_group = payload.logGroup;
-        data.platform_log_stream = payload.logStream;
-        data.timestamp = new Date(1 * logEvent.timestamp).toISOString();
-        let message = logEvent.message;
-        let messageLength = Buffer.byteLength(message, 'utf8');
-        if (messageLength > 32766) { //since 32766(32KB) is the default message size
-          let truncatedMessage = truncate(message, 32740); // message size + ...[TRUNCATED]
-          data.message = truncatedMessage + "  ...[TRUNCATED]";
+    data.asset_type = "lambda";
+    data.provider = "aws";
+    let domainAndservice, serviceInfo, environment, dev_environment, serviceInfoArr, domain;
+    data.request_id = getInfo(payload.logEvents, config.PATTERNS.lambda_request_id);
+    if (data.request_id) {
+      data.asset_identifier = payload.logGroup.split(config.PATTERNS.asset_identifier_key)[1];
+      // if tags present, then it is for sls-app, otherwise for other services
+      if(tagsResult != 'error' && tagsResult != undefined && tagsResult.tags.environment){
```

Comment:
Check for `tagsResult` first & then the error.. In fact, below should suffice..

` if(tagsResult && tagsResult.tags) {..}`

Reply:
Actually the output of getLogsGroupsTags function will be either 'error' or 'tags:{...}'. And even if the logGroup doesn't have tags, the output will be empty tags 'tags:{}'. So just depending on 'tagsResult.tags' won't work. That's why the first check is for 'error' and then checking for 'tags.environment', since 'tags' keyword will be there irrespective of the logGroup having tags or not. 

Will remove the undefined check